services/src/main/java/com/chatbot/services/pubsubservices/PubSubController.java

```diff
@@ -0,0 +1,125 @@
+package com.chatbot.services.pubsubservices;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Map;
+
+import com.chatbot.services.ChatServiceConstants;
+import com.chatbot.services.asyncservices.HangoutsAsyncService;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.ChatClient;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.Event;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.protobuf.Struct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+
+// Handle messages from cloud sub using a push subscriber
+@RestController
+public class PubSubController {
+
+  private HangoutsAsyncService asyncService;
+  private PubSubAuth pubSubAuth;
```

Comment:
private final

Reply:
Injected variables can't be final

Reply:
They can be if they are injected in the constructor.

Is this some spring boot related thing? If yes, can you give me reference to some error or a document

Reply:
Not sure, but adding final is breaking the tests